src/wasm.rs

```diff
@@ -141,12 +134,38 @@ fn sha256_digest<R: Read>(mut reader: R) -> Result<Digest> {
     Ok(context.finish())
 }
 
-fn compute_hash_without_jwt(module: Module) -> Result<String> {
-    let mut refmod = module;
-    refmod.clear_custom_section("jwt");
-    let modbytes = serialize(refmod)?;
+// NOTE: we don't need to compute a hash of the entire file, we just need
+// to compute the hash if the things that indicate tampering, like code and
+// custom sections
```

Comment:
Our rule has always been to sign the file after post processing for this very reason.

Further, because of the nature of wasm, it's impossible to execute code that isn't in a code section and this algorithm hashes all code and data.

We don't know the binary format for components yet so there's nothing we can do here. The component signing algorithm may end up being entirely different.